specific_use_cases/DFU_work_in_progress/station_8760s.ipynb

```diff
@@ -0,0 +1,627 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aab21ce6-148a-4408-9d41-2f3e7727a136",
+   "metadata": {},
+   "source": [
+    "Finally, the heatmap:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "65ea17fd-bba4-4a31-95a0-014065b4b7f3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "median_year_array.hvplot(\n",
+    "    x='dayofyear',\n",
+    "    y='hour',\n",
+    "    yticks=[(int(one_hour),to_PST(one_hour)) for one_hour in median_year_array.hour.values],\n",
+    "    groupby='station',\n",
+    "    cmap='Reds'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c4c0141a-cd82-4031-a681-fec622cb79e1",
+   "metadata": {},
+   "source": [
+    "### Export\n",
+    "We'll show exporting of the data as originally accessed, but you could also export the processed data in a similar fashion.\n",
+    "#### netCDF\n",
+    "The simplest way to export the data is to use the netCDF format, which this data structure already mirrors in memory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bb009cff-70bd-4645-b6f5-ddaa4490c051",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data.to_netcdf('test_output.nc')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94aead2e-a2ed-47f1-bf3a-449aa8103395",
+   "metadata": {},
+   "source": [
+    "The file should appear in the file list to the left, after the cell is done executing $[*]$.\n",
+    "\n",
+    "For reference, to read this back in using python:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e90dfe46-c5a1-4f74-a76f-0c5422876edc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import xarray as xr\n",
+    "data_back = xr.open_dataset('test_output.nc')\n",
+    "data_back"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "14de0ed8-6709-44ca-96b5-76b293a0f6f5",
+   "metadata": {},
+   "source": [
+    "#### csv\n",
+    "If for some reason you prefer a .csv file, first convert the data structure to a table compatible with export to csv."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0de49cd3-47b5-4021-a062-d6a330978a91",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_table = data.to_dataframe(dim_order=['scenario','simulation','time'])\n",
+    "data_table"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1cc0d82b-9a01-47c2-b975-6e5638f7ca83",
+   "metadata": {},
+   "source": [
+    "You can save the above out to csv, but it will be too many rows for e.g. Excel to read in.\n",
+    "\n",
+    "We can also rearrange the data so it's distributed over more columns instead, although it may take a moment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d9e1c694-2c40-4db7-9fba-8219306ae21a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_table = data_table.unstack().unstack().stack('time')"
```

Comment:
This structure is not ideal for reading into R for a few reasons - 
1) the nested header columns can't be easily represented in R data frame format (preference is for the structure shown in the cell above, L525, but with the scenario and simulation column values repeated for each row they apply to - this is the more standard way to work with csvs/'tidy' data in R which I know the DFU folks do)
2) spaces in column headers is a real pain in R as well - preference for snake or camel case if possible!

Reply:
Please confirm that this is better.

Reply:
Hmm this version still reads out with some nesting in the column headers. I think the version we should recommend them to export is the one earlier in the notebook, in the chunk that reads:

`data_table = data.to_dataframe(dim_order=['scenario','simulation','time'])`

I checked that one in R, and although the preview of the data table in Jupyter does not repeat the scenario & simulation row values (which I was concerned about), the exported CSV does. So, I'd suggest just adding one more line to remove spaces in the column names (this is what I would do but maybe there's another way that's preferred):

`data_table.columns = data_table.columns.str.replace(' ', '_')`

And then doing the export to csv step for that file. The size is not an issue for reading into R, but we can keep the note about it being too large to open in Excel.

Reply:
Okay good to know. Will update as suggested.